nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-client-service-api/src/main/java/org/apache/nifi/distributed/cache/client/AtomicDistributedMapCacheClient.java

```diff
@@ -71,6 +73,23 @@
      * @return true only if the key is replaced.
      * @throws IOException if unable to communicate with the remote instance
      */
+    @Deprecated
     <K, V> boolean replace(K key, V value, Serializer<K> keySerializer, Serializer<V> valueSerializer, long revision) throws IOException;
```

Comment:
@ijokarumawak I think using generics for the revision should work, I had actually considered this idea and not really sure why I didn't go this route, but I am open to giving it a try.

We would also need to pass in a Serializer<R> revisionSerializer to the replace method right?

So the process would be... fetch the existing CacheEntry, update the value (V), and pass it back in with the serializers.

I'll give it a try in the next couple of days, thanks for taking a look!